File: dev/test_data/main/csharp/expected/implementation_specific/input/snippets/Verification/items_have_unique_labels.cs
/// <summary>
/// Check that <see cref="Aas.IItem.Label" />'s of the <paramref name="items" />
/// do not repeat.
/// </summary>
public static bool ItemsHaveUniqueLabels(IEnumerable<Aas.IItem> items)
{
    var labelSet = new HashSet<string>();
    foreach (var item in items)
    {
        if (labelSet.Contains(item.Label))
        {
            return false;
        }
        labelSet.Add(item.Label);
    }
    return true;
}

File: dev/test_data/main/golang/expected/implementation_specific/input/snippets/Verification/items_have_unique_labels.go
// Check that [aastypes.IItem.Label]'s of the `items` do not repeat.
func ItemsHaveUniqueLabels[I aastypes.IItem](items []I) bool {
	labelSet := make(map[string]struct{})

	for _, item := range items {
		label := item.Label()
		_, has := labelSet[label]
		if has {
			return false
		}

		labelSet[label] = struct{}{}
	}
	return true
}
